### PR TITLE
Fix Windows compatibility in configure.py

### LINKED ISSUE #139: configure.py -- chmod does not work in Windows
`chmod` is not a valid command in a Windows shell, so the following snippet fails under Windows:

https://github.com/rice-solar-physics/pydrad/blob/bc1e0e74f02f9d714f36d7412723a1a3c3c4db3d/pydrad/configure/configure.py#L129-L133

```
C:\ProgramData\Anaconda3\lib\site-packages\pydrad\configure\configure.py in setup_simulation(self, output_path, base_path, run_initial_conditions, **kwargs)
     87             if run_initial_conditions:
     88                 execute = kwargs.get('execute', True)
---> 89                 self.setup_initial_conditions(tmpdir, execute=execute)
     90             self.setup_hydrad(tmpdir)
     91             self.save_config(os.path.join(tmpdir, 'pydrad_config.asdf'))

C:\ProgramData\Anaconda3\lib\site-packages\pydrad\configure\configure.py in setup_initial_conditions(self, root_dir, execute)
    127                 f.write(filestring)
    128         # NOTE: make sure we have needed permissions to run compile script
--> 129         run_shell_command(
    130             ['chmod', 'u+x', 'build_initial_conditions.bat'],
    131             os.path.join(root_dir, 'Initial_Conditions/build_scripts'),

C:\ProgramData\Anaconda3\lib\site-packages\pydrad\configure\util.py in run_shell_command(cmd, cwd, shell)
     22 
     23 def run_shell_command(cmd, cwd, shell=True):
---> 24     cmd = subprocess.run(
     25         cmd,
     26         cwd=cwd,

C:\ProgramData\Anaconda3\lib\subprocess.py in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    487         kwargs['stderr'] = PIPE
    488 
--> 489     with Popen(*popenargs, **kwargs) as process:
    490         try:
    491             stdout, stderr = process.communicate(input, timeout=timeout)

C:\ProgramData\Anaconda3\lib\subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors, text)
    852                             encoding=encoding, errors=errors)
    853 
--> 854             self._execute_child(args, executable, preexec_fn, close_fds,
    855                                 pass_fds, cwd, env,
    856                                 startupinfo, creationflags, shell,

C:\ProgramData\Anaconda3\lib\subprocess.py in _execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, unused_restore_signals, unused_start_new_session)
   1305             # Start the process
   1306             try:
-> 1307                 hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
   1308                                          # no special security
   1309                                          None, None,

FileNotFoundError: [WinError 2] The system cannot find the file specified
```

Similarly a bit further down:
https://github.com/rice-solar-physics/pydrad/blob/bc1e0e74f02f9d714f36d7412723a1a3c3c4db3d/pydrad/configure/configure.py#L203-L207

Likely this needs a check for the operating system.

### PR DESCRIPTION
**Pull request by jwreep**
Fixes #139

Fixes the use of `chmod` and `.`, both of which are Unix-based commands and invalid on Windows.  I tested this on Windows 10 and then verified on MacOS that it doesn't break anything.

### CODE REVIEW

**Comment by wtbarnes** [comment]
The `chmod` conditional looks good to me. As far as prepending a `./` goes, I wonder if this logic could just go in the `run_shell_command` script? I'm just reluctant to make the `setup_intitial_conditions` and `setup_hydrad` methods even more hard to follow than they already are and moving that logic to a separate function seems preferable.

So for example, you could just have a check for Windows and then if the command starts with a `./` and you're on windows, strip those characters.

File: pydrad/configure/configure.py [hunk 1]
```diff
@@ -126,23 +127,36 @@ def setup_initial_conditions(self, root_dir, execute=True):
...
-            ['chmod', 'u+x', 'build_initial_conditions.bat'],
-            os.path.join(root_dir, 'Initial_Conditions/build_scripts'),
-            shell=False
-        )
-        run_shell_command(
-            ['./build_initial_conditions.bat'],
-            os.path.join(root_dir, 'Initial_Conditions/build_scripts')
-        )
+		# Only do this on Unix-based systems
+        if platform.system() != 'Windows':
```
Comment (wtbarnes) [inline pydrad/configure/configure.py:131]: Since this is used multiple times, I would actually consider making it a separate property on `Configure`,

```python
@property
def on_windows(self):
    return platform.system().lower() == 'windows'
```

**Comment by jwreep** [comment]
I've changed it as suggested.  It appears to be working on windows.  The one concern is that the modification to util.py is not terribly clean -- there's probably an easier-to-read way to write `cmd = [cmd[0][2:]]` to remove the leading `./`, but it gets the job done.  Any suggestion there?

File: pydrad/configure/util.py [hunk 2]
```diff
@@ -21,6 +22,9 @@ class HYDRADError(Exception):
 
 
 def run_shell_command(cmd, cwd, shell=True):
+    # remove "./" from commands if the user is working on Windows
+    if platform.system().lower() == 'windows' and cmd[0][0:2] == './':
+        cmd = [cmd[0][2:]]
```
Comment (wtbarnes) [inline pydrad/configure/util.py:27]: You should only need to alter the first entry
```suggestion
        cmd[0] = cmd[0][2:]
```

File: pydrad/configure/configure.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import datetime
 import tempfile
 import shutil
+import platform
```
Comment (wtbarnes) [inline pydrad/configure/configure.py:9]: Don't need this import anymore
```suggestion
```

File: pydrad/configure/configure.py [hunk 4]
```diff
@@ -126,14 +127,16 @@ def setup_initial_conditions(self, root_dir, execute=True):
...
         run_shell_command(
-            ['chmod', 'u+x', 'build_initial_conditions.bat'],
-            os.path.join(root_dir, 'Initial_Conditions/build_scripts'),
-            shell=False
-        )
-        run_shell_command(
-            ['./build_initial_conditions.bat'],
-            os.path.join(root_dir, 'Initial_Conditions/build_scripts')
+			['./build_initial_conditions.bat'],
+			os.path.join(root_dir, 'Initial_Conditions/build_scripts')
```
Comment (wtbarnes) [inline pydrad/configure/configure.py:139]: I think the indentation got messed up here.
```suggestion
		['./build_initial_conditions.bat'],
		os.path.join(root_dir, 'Initial_Conditions/build_scripts')
```

File: pydrad/configure/configure.py [hunk 5]
```diff
@@ -200,14 +203,16 @@ def setup_hydrad(self, root_dir):
...
         run_shell_command(
-            ['chmod', 'u+x', build_script],
-            os.path.join(root_dir, 'HYDRAD/build_scripts'),
-            shell=False,
-        )
-        run_shell_command(
-            [f'./{build_script}'],
-            os.path.join(root_dir, 'HYDRAD/build_scripts'),
+			[f'./{build_script}'],
+			os.path.join(root_dir, 'HYDRAD/build_scripts'),
```
Comment (wtbarnes) [inline pydrad/configure/configure.py:215]: And here too
```suggestion
		[f'./{build_script}'],
		os.path.join(root_dir, 'HYDRAD/build_scripts'),
```

**Review by wtbarnes — CHANGES_REQUESTED** [review]
As far as I know there's not really a cleaner way to handle stripping the `./` stuff. Maybe the subprocess module provides some way to run commands in a more OS-safe way? I think our hacky approach is probably good enough for now.

File: pydrad/configure/configure.py (hunk 3 above)
Comment (jwreep) [reply pydrad/configure/configure.py:9]: It's used in the `on_windows` function, unless you think that could be moved to util.py?

File: pydrad/configure/configure.py (hunk 5 above)
Comment (jwreep) [reply pydrad/configure/configure.py:215]: Probably because of the editor I was using, which prefers tabs to four spaces.

File: pydrad/configure/configure.py (hunk 3 above)
Comment (jwreep) [reply pydrad/configure/configure.py:9]: That might actually be cleaner since it is used again in util.py...

Comment (wtbarnes) [reply pydrad/configure/configure.py:9]: You only need the import in the file it's explicitly used, in this case `util.py`

Comment (jwreep) [reply pydrad/configure/configure.py:9]: The `on_windows` function is in configure.py currently, so the import is needed.  What I meant is that perhaps `on_windows` should be moved to util.py.

Comment (wtbarnes) [reply pydrad/configure/configure.py:9]: Sorry, I don't know what I'm thinking. 🤦‍♂️ You're right. 

Actually though, I think it would make more sense to move that function to `util.py` since it gets used in `run_shell_command`

**Review by wtbarnes — APPROVED** [review]

**Comment by wtbarnes** [comment]
> Right, I've moved `on_windows` to util.py. I tested on Windows without issue. It might be worth double-checking on Unix, too.

I just made a few last minor changes and tested on Linux and it all looks good!
